presto-main/src/main/java/io/prestosql/operator/GroupedTopNRankBuilder.java

```diff
@@ -93,56 +120,41 @@ public long getEstimatedSizeInBytes()
         return INSTANCE_SIZE
                 + groupByHash.getEstimatedSize()
                 + pageManager.sizeOf()
-                + groupedTopNRowNumberAccumulator.sizeOf();
+                + groupedTopNRankAccumulator.sizeOf();
     }
 
     private void processPage(Page newPage, GroupByIdBlock groupIds)
     {
         try (LoadCursor loadCursor = pageManager.add(newPage)) {
-            GroupedTopNRowNumberAccumulator.RowReference rowReferenceView = asRowReferenceView(loadCursor);
             for (int position = 0; position < newPage.getPositionCount(); position++) {
                 long groupId = groupIds.getGroupId(position);
                 loadCursor.advance();
-                groupedTopNRowNumberAccumulator.add(groupId, rowReferenceView);
+                long rowId = loadCursor.allocateRowId();
```

Comment:
Is it possible to optimize this loop to avoid allocation of row id?

Reply:
You'll notice that we did this optimization in the topNRowNumber because it was possible. I didn't do it here because we hit the LongLong map first, which makes it pointless. I have some downstream changes that might improve this, but it is quite destructive to a lot of this structure, so I didn't want to put that in quite yet.